vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/Map.java

```diff
@@ -70,6 +80,46 @@ public class Map extends MapBase {
     private Layer backgroundLayer;
     private final FeatureLayer featureLayer;
 
+    /**
+     * Sets the projection (or coordinate system) to use for all coordinates.
+     * That means that all coordinates passed to, or returned from the public
+     * API, must be in this projection. Internally the coordinates will be
+     * converted into the projection that is used by the map's {@link View}.
+     * <p>
+     * By default, the user projection is set to {@code EPSG:4326}, also known
+     * as latitude / longitude, or GPS coordinates.
+     * <p>
+     * This setting affects all maps in the current {@link UI}, currently it is
+     * not possible to configure this per map instance. This method may only be
+     * invoked inside of UI threads, and will throw otherwise. This setting
+     * being scoped to the current UI means that it will stay active when
+     * navigating between pages using the Vaadin router, but not when doing a
+     * "hard" location change, or when reloading the page. As such it is
+     * recommended to apply this setting on every page that displays maps. Note
+     * that when using the preserve on refresh feature, a view's constructor is
+     * not called. In that case this setting can be applied in an attach
+     * listener.
+     * <p>
+     * This method should be called before creating any maps. Changing this
+     * setting does not affect existing maps, specifically the component does
+     * not convert coordinates configured in an existing map into the new
+     * projection. Instead, existing maps should be recreated after changing
+     * this setting.
+     *
+     * @param projection
+     *            the user projection to use for all public facing API
+     */
+    public static void setUserProjection(String projection) {
+        UI ui = UI.getCurrent();
+        if (ui == null || ui.getPage() == null) {
+            throw new IllegalStateException(
+                    "Setting a user projection requires a current UI, and a page.");
```

Comment:
Consider using a bit more detailed error description to align with other components, example:

https://github.com/vaadin/flow-components/blob/48830df666083a2ee34e625daa1ca3f7f40489a3/vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java#L734-L738

I'm also not sure if `ui.getPage()` can be `null` as we only check for `ui == null` elsewhere.

Reply:
Updated the exception message. Let's keep the check for the page, in other parts of the code we don't necessarily use the page instance, and the check doesn't really hurt.